lib/async/enumerable.rb

```diff
@@ -22,15 +23,19 @@ module Async
   # Provides async parallel execution for Enumerable.
   # See docs/reference/enumerable.md for detailed documentation.
   module Enumerable
-    DEFAULT_MAX_FIBERS = 1024
-
     class << self
-      attr_writer :max_fibers
+      attr_accessor :config
```

Comment:
let's make config a method that optionally accepts kwargs, creating/memoizing a new config object with the optional kwargs if one does not exist. -- this will remove most of the logic from the `max_fibers` and `max_fibers=` methods allowing them to just refer to `config.max_fibers` or in the case of the setter, just call `config(**kwargs).max_fibers`